clippy_lints/src/matches.rs

```diff
@@ -415,7 +414,7 @@ fn check_match_ref_pats(cx: &LateContext, ex: &Expr, arms: &[Arm], source: Match
 }
 
 /// Get all arms that are unbounded `PatRange`s.
-fn all_ranges(cx: &LateContext, arms: &[Arm]) -> Vec<SpannedRange<ConstVal>> {
+fn all_ranges<'a>(cx: &'a LateContext, arms: &'a [Arm]) -> Vec<SpannedRange<ConstVal<'a>>> {
```

Comment:
that's the wrong lifetime. You need an additional `'tcx` lifetime and use it in `LateContext<'tcx>` and `ConstVal<'tcx>`, no need for a lifetime in `arms`